analytics/analytics_test.go

```diff
@@ -0,0 +1,45 @@
+package analytics
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_initData(t *testing.T) {
+	tests := []struct {
+		name       string
+		data, want map[string]interface{}
+	}{
+		{
+			name: "Empty data",
+			data: map[string]interface{}{},
+			want: map[string]interface{}{
+				"source": "scanner",
+			},
+		},
+		{
+			name: "nil data",
+			data: nil,
+			want: map[string]interface{}{
+				"source": "scanner",
+			},
+		},
+		{
```

Comment:
NOTE: might a test case which represents that we set (overwrite the existing value of) `source` would be useful.